### ISSUE TITLE
Payments Task List Epic (part of WCPay overview screen v2)

### ISSUE THREAD

**Issue by samueljseay** [Automattic/woocommerce-payments#1655]
This epic covers the work required to implement the new task list in the WCPay overview v2. See paJDYF-18j-p2 for the designs and further information.

I'm suggesting that we do this work behind a feature flag because the overview screen is slated to be the default screen once https://github.com/Automattic/woocommerce-payments/issues/1412 is complete and there are quite a lot of tasks involved to implement the task list.

We could also start by implementing 1 or 2 of the initial task list items before going live, or if its preferred we can feature flag the whole task list till the initial tasks are ready. @pmcpinto wondering if you have any opinions on this.

Ideally we would use new components being developed in `@woocommerce/components` to create a task list in WCPay. In that case ideally we'd have the following tasks done before starting this:

https://github.com/woocommerce/woocommerce-admin/issues/6772
https://github.com/woocommerce/woocommerce-admin/issues/6639
https://github.com/woocommerce/woocommerce-admin/issues/6636

**Comment by samueljseay** [comment]
> We could also start by implementing 1 or 2 of the initial task list items before going live, or if its preferred we can feature flag the whole task list till the initial tasks are ready. @pmcpinto wondering if you have any opinions on this.

@pmcpinto pinging you on this again and cc @elizaan36 @timmyc .

Any opinions on how we should roll this out gradually?

**Comment by pmcpinto** [comment]
@sam and I spoke about this a couple of days ago. We can't easily extract the UI that was built in WC Admin into WC Payments. So the best option is to assess the effort required to implement a trimmed-down version of what we have in WC Admin. Sam, please correct me if I'm wrong :)

**Comment by samueljseay** [comment]
@jeffstieler @louwie17 where do you think we're at with bringing the task list into wc pay? It looks like https://github.com/Automattic/woocommerce-payments/pull/1772 was merged, but that is just a placeholder, that doesn't actually implement https://github.com/automattic/woocommerce-payments/issues/1656 in its entirety?

**Comment by jeffstieler** [comment]
@samueljseay what's missing from #1656? That issue said to implement a placeholder list - the individual tasks were in other issues:

* https://github.com/Automattic/woocommerce-payments/pull/1799 added the "update business details" task from https://github.com/automattic/woocommerce-payments/issues/1657
* The "dispute resolution" task is in https://github.com/automattic/woocommerce-payments/issues/1658
* Integrating the above tasks with the WC Admin homescreen task list is in https://github.com/automattic/woocommerce-payments/issues/1661

**Comment by samueljseay** [comment]
@jeffstieler apologies I think the word placeholder made this misleading for me. It's not a placeholder but the empty list? In that case it's all good 👍

**Comment by mattsherman** [comment]
I'm going to move this epic over to Mothra, since the only remaining issue is being handled by a Mothra team member.

**Comment by mattsherman** [comment]
Closing this epic. Spun the dispute task work out into a separate epic to allow for better tracking (in retrospect, this should have been done earlier).

https://github.com/Automattic/woocommerce-payments/issues/2400
